models/repo.go

```diff
@@ -2806,6 +2810,64 @@ func GenerateRepository(doer, owner *User, templateRepo *Repository, opts Genera
 		}
 	}
 
+	// Git Hooks
+	if opts.GitHooks {
+		generateRepo, err := git.OpenRepository(repo.repoPath(sess))
+		if err != nil {
+			return repo, err
+		}
+
+		templateRepo, err := git.OpenRepository(templateRepo.repoPath(sess))
+		if err != nil {
+			return repo, err
+		}
+
+		templateHooks, err := templateRepo.Hooks()
+		if err != nil {
+			return repo, err
+		}
+
+		for _, templateHook := range templateHooks {
+			generateHook, err := generateRepo.GetHook(templateHook.Name())
+			if err != nil {
+				return repo, err
+			}
+
+			generateHook.Content = templateHook.Content
+			generateHook.IsActive = templateHook.IsActive
```

Comment:
```suggestion
```
What is this for? `generateHook.Update()` will set it to true, anyway.

Reply:
Ah, good catch.